samza-sql/src/main/java/org/apache/samza/sql/translator/JoinTranslator.java

```diff
@@ -59,46 +60,53 @@
  * Translator to translate the LogicalJoin node in the relational graph to the corresponding StreamGraph
  * implementation.
  * Join is supported with the following caveats:
- *   0. Only local tables are supported. Remote/composite tables are not yet supported.
  *   1. Only stream-table joins are supported. No stream-stream joins.
  *   2. Only Equi-joins are supported. No theta-joins.
  *   3. Inner joins, Left and Right outer joins are supported. No cross joins, full outer joins or natural joins.
  *   4. Join condition with a constant is not supported.
  *   5. Compound join condition with only AND operator is supported. AND operator with a constant is not supported. No
  *      support for OR operator or any other operator in the join condition.
- *
- * It is assumed that the stream denoted as 'table' is already partitioned by the key(s) specified in the join
- * condition. We do not repartition the table as bootstrap semantic is not propagated to the intermediate streams.
- * Please refer SAMZA-1613 for more details on this. But we always repartition the stream by the key(s) specified in
- * the join condition.
+ * For local table, we always repartition both the stream to be joined and the stream denoted as table by the key(s)
+ * specified in the join condition.
  */
 class JoinTranslator {
 
   private static final Logger log = LoggerFactory.getLogger(JoinTranslator.class);
   private int joinId;
-  private SqlIOResolver ioResolver;
   private final String intermediateStreamPrefix;
+  private final int queryId;
+  enum TableType {
+    NONE, // Not a table == a stream
```

Comment:
Why do we have a TableType which is not a table but a stream?

Reply:
Join has two inputs to it. And the type of input could be either a stream, local table or remote table. And that is what I wanted to represent here in the enum. Should I change the enum name to InputType and rename the values to STREAM, LOCAL_TABLE, REMOTE_TABLE ? Splitting them as a boolean for stream or table and then another boolean for local or remote would be another approach.

Reply:
I refactored this part of the code. Please take a look.